backend/migrations/0004_remove_username.down.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE fluke_user
+ADD COLUMN username TEXT NOT NULL UNIQUE;
```

Comment:
won't this cause an error creating a "not null" column on a db with no entries for this? add default value of something? not entirely sure what to do here

Reply:
Adding a default value would violate the unique constraint, so it needs to be different for each row. Seems like we could either 

1. Make a migration to first get rid of UNIQUE 
2. Do something like this: 

`alter TABLE users ADD column register_key text NOT NULL default substring(md5(random()::text), 0, 8)`
From [this stack exchange question](https://dba.stackexchange.com/questions/19632/how-to-create-column-in-db-with-default-value-random-string)

..or just keep the username. Holding off the merge until we agree on something

Reply:
Yeah so this migration does break, it leads to `error: while executing migrations: error returned from database: column "username" of relation "fluke_user" contains null values`

The above migration allows a proper revert, as well as this one I tested: 

```SQL
ALTER TABLE fluke_user
ADD COLUMN username TEXT;

UPDATE fluke_user
SET username = 'user_' || id;

ALTER TABLE fluke_user
ALTER COLUMN username SET NOT NULL;

ALTER TABLE fluke_user
ADD CONSTRAINT username_unique UNIQUE (username);
```

I like the logic of the second version ever, user_id makes sense for a unique username

Reply:
Just remove the `NOT NULL`?

Reply:
Or populate the username with the email address by default?

Reply:
email address worked for me, see a1625ce42be56d6fd69212b21fcf3796a889f2ab